app/controllers/katello/api/v2/packages_controller.rb

```diff
@@ -55,7 +55,7 @@ def custom_index_relation(collection)
       upgradable = ::Foreman::Cast.to_bool(params[:packages_restrict_upgradable])
       if applicable || upgradable
         hosts = @host ? [@host] : ::Host::Managed.authorized("view_hosts")
```

Comment:
The way this code is organized seems unnecessarily confusing. It's not because of your change and I know you're trying to make a quick fix. A way to clarify this code overall could be to refactor `find_host` into `find_hosts`:

```
    def find_hosts                                                                                                                                                                            
      @hosts = ::Host::Managed.authorized("view_hosts")                                                                                                                                       
      if params[:host_id]                                                                                                                                                                     
        @hosts = @hosts.where(:id => params[:host_id])                                                                                                                                        
        fail HttpErrors::NotFound, _('Could not find a host with id %s') % params[:host_id] if @hosts.empty?                                                                                  
      end                                                                                                                                                                                     
      @hosts = @hosts.where(:organization_id => params[:organization_id]) if params[:organization_id]                                                                                         
    end
```

then the `custom_index_relation` becomes
```
    def custom_index_relation(collection)                                                                                                                                                     
      applicable = ::Foreman::Cast.to_bool(params[:packages_restrict_applicable]) || @hosts.any?                                                                                                     
      upgradable = ::Foreman::Cast.to_bool(params[:packages_restrict_upgradable])                                                                                                             
                                                                                                                                                                                              
      if applicable || upgradable                                                                                                                                    
        if upgradable                                                                                                                                                                         
          collection = collection.installable_for_hosts(@hosts)                                                                                                                               
        elsif applicable                                                                                                                                                                      
          collection = collection.applicable_to_hosts(@hosts)                                                                                                                                 
        end                                                                                                                                                                                   
      end                                                                                                                                                                                     
      collection                                                                                                                                                                              
    end
```

I think the end result is the same and to me the code is a lot easier to understand

Reply:
I'll leave it up to you as to whether you want to make this change, but I do think it's an improvement especially because it will always return a relation for `@hosts` which is generally useful!

Reply:
Looks like it should work well! I'll try it out, always like an improvement.